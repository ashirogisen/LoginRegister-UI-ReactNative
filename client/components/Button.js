import React from 'react'
import { StyleSheet } from 'react-native'
import { Button, Text } from 'native-base'

function Buttons({ onSubmit, name, customStyles }) {
  const styles = StyleSheet.create({
    button: {
      width: '100%',
      justifyContent: 'center',
      alignItems: 'center'
    }
  })

  return (
    <Button 
      testID="test-button" 
      onPress={onSubmit} 
      rounded={true}
      style={{ ...styles.button, ...customStyles }}
    >
      <Text testID="test-buttonText">{name}</Text>
    </Button>
  )
}

export default Buttons
