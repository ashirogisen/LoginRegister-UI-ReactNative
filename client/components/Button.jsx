import React from 'react';
import { Container, Header, Content, Button, Text } from 'native-base';

function Buttons(props) {
  return (
    <Text>{props.name}</Text>
  )
}

export default Buttons
