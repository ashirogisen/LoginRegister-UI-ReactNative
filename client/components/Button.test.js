import React from 'react'
import Button from './Button'
import { render, fireEvent } from 'react-native-testing-library'

const handleSubmitMocked = jest.fn()

function renderButton(props) {
  const component = render (
    <Button {...props} />
  )
  const button = component.queryByTestId('test-button')
  const buttonText = component.queryByTestId('test-buttonText')

  return {
    ...component,
    button,
    buttonText
  }
}

describe('Render Button', () => {
  test('Renders without crashing', () => {
    const component = renderButton()
    expect(component).toMatchSnapshot()
  })

  test('Should render buttons body, and text', () => {
    const { button, buttonText } = renderButton()
    expect(button).toBeTruthy()
    expect(buttonText).toBeTruthy()
  })

  test('Function can be called', () => {
    const { button } = renderButton({ onSubmit: handleSubmitMocked })
    fireEvent.press(button)
    expect(handleSubmitMocked).toHaveBeenCalledTimes(1)
  })
})
