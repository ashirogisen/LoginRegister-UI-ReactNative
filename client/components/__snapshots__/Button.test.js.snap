// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button Renders without crashing 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#007aff",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 57,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "borderWidth": undefined,
      "elevation": 2,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": undefined,
      "shadowOffset": undefined,
      "shadowOpacity": undefined,
      "shadowRadius": undefined,
      "width": "100%",
    }
  }
  testID="test-button"
>
  <Text
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#fff",
        "fontFamily": "System",
        "fontSize": 16.5,
        "marginLeft": 0,
        "marginRight": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
    testID="test-buttonText"
    uppercase={false}
  />
</View>
`;
